Services/Simulations.cs

```diff
@@ -192,7 +192,19 @@ public async Task<Models.Simulation> UpsertAsync(Models.Simulation simulation)
                 throw new InvalidInputException("Simulation ID is not specified.");
             }
 
-            var existingSimulation = await this.GetAsync(simulation.Id);
+            Models.Simulation existingSimulation = null;
```

Comment:
If the simulation ID doesn't exist, the caller should be calling InsertAsync, not UpsertAsync. Perhaps we should leave this unchanged and determine why UpsertAsync is being called on a simulation that doesn't exist.